Shared/Samples/Edit with branch versioning/EditWithBranchVersioningView.swift

```diff
@@ -0,0 +1,258 @@
+                    model.selectedFeature?.setAttributeValue(
+                        damageType.rawValue,
+                        forKey: .damageTypeKey
+                    )
+                    selectedAction = .updateFeature
+                }
+            }
+        } message: {
+            Text("Choose a damage type for the building.")
+        }
+    }
+    
+    /// The button for creating a version.
+    private var createVersionButton: some View {
+        Button("Create") {
+            isCreatingVersion = true
+        }
+        .popover(isPresented: $isCreatingVersion) {
+            CreateVersionParametersView(model: model) { parameters in
+                // Makes a new version with the created parameters.
+                versionParameters = parameters
+                selectedAction = .makeVersion
+            }
+            .presentationDetents([.fraction(0.5)])
+            .frame(idealWidth: 320, idealHeight: 200)
+        }
+    }
+    
+    /// The button for switching the current version.
+    private var switchVersionButton: some View {
+        Button("Switch") {
+            isSwitchingVersion = true
+        }
+        .disabled(model.existingVersionNames.count < 2)
+        .confirmationDialog("Versions", isPresented: $isSwitchingVersion, titleVisibility: .visible) {
+            ForEach(model.existingVersionNames, id: \.self) { versionName in
+                Button(versionName) {
+                    selectedAction = .switchToVersion(version: versionName)
+                }
+            }
+        } message: {
+            Text("Choose a version to switch to.")
+        }
+    }
+}
+
+/// An asynchronous action associated with the sample.
+private enum AsyncAction: Equatable {
+    /// Sets up the sample.
+    case setUp
+    
+    /// Makes a version with the current version parameters.
+    case makeVersion
+    
+    /// Switches to a version with an associated version name.
+    case switchToVersion(version: String)
+    
+    /// Select a feature identified from an associated screen and map point.
+    case selectFeature(screenPoint: CGPoint, mapPoint: Point)
+    
+    /// Updates the selected feature in it's feature table.
+    case updateFeature
+}
+
+/// The damage type of a feature.
+private enum DamageType: String, CaseIterable {
+    case destroyed, major, minor, affected, inaccessible, `default`
+    
+    /// A human-readable label for the damage type.
+    var label: String {
+        switch self {
+        case .destroyed: "Destroyed"
+        case .major: "Major"
+        case .minor: "Minor"
+        case .affected: "Affected"
+        case .inaccessible: "Inaccessible"
+        case .default: "Default"
+        }
+    }
+}
+
+private extension String {
+    /// The key for a feature's damage type attribute.
+    static let damageTypeKey = "typdamage"
+    
+    /// The key for a feature's place name attribute.
+    static let placeNameKey = "placename"
```

Comment:
We don't usually separate stored properties with a blank line.

```suggestion
    /// The key for a feature's damage type attribute.
    static let damageTypeKey = "typdamage"
    /// The key for a feature's place name attribute.
    static let placeNameKey = "placename"
```